app/src/main/java/com/bps/gotwinter2021/favorites/database/FavoriteDatabaseDao.kt

```diff
@@ -0,0 +1,36 @@
+package com.bps.gotwinter2021.favorites.database
+
+import androidx.lifecycle.LiveData
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+import androidx.room.Update
+
+@Dao
+interface FavoriteDatabaseDao {
+    @Insert
+    fun insert(favorite: Favorite)
+
+    @Update
+    fun update(favorite: Favorite)
+
+    //find a specific character based on its name
+    @Query("SELECT * from favorite_characters_table WHERE character_name = :key")
+    fun findCharacter(key: String) : Favorite
```

Comment:
you should add the question mark "?" in the : Favorite? in case there are no favorites